File: Numerical_Methods/Numerical Solutions of Algebraic Equations/Newton-Raphson Method/Java_Solutions/Problem10.java
public class Problem10 {
    public static void main(String[] args) {
        double x0 = 4.5;  // Initial guess
        double tolerance = 0.0001;
        int maxIterations = 10;
        
        System.out.println("Newton-Raphson Method for f(x) = tan(x) - x = 0");
        System.out.println("Initial guess: x₀ = " + x0);
        System.out.println("\nIterations:");
        
        double x = x0;
        for (int i = 0; i < maxIterations; i++) {
            double fx = Math.tan(x) - x;           // f(x) = tan(x) - x
            double fpx = 1.0/Math.pow(Math.cos(x), 2) - 1;  // f'(x) = sec²(x) - 1
            
            double xNext = x - fx / fpx;
            
            System.out.printf("Iteration %d: x = %.6f, f(x) = %.6f\n", 
                            i + 1, x, fx);
            
            if (Math.abs(xNext - x) < tolerance) {
                System.out.printf("\nRoot found: %.6f\n", xNext);
                break;
            }
            
            x = xNext;
        }
    }
} 

File: Numerical_Methods/Numerical Solutions of Algebraic Equations/Newton-Raphson Method/Python_Solutions/Problem10.py
import math

def newton_raphson(x0, tolerance=0.0001, max_iterations=10):
    x = x0
    print(f"Newton-Raphson Method for f(x) = tan(x) - x = 0")
    print(f"Initial guess: x₀ = {x0}")
    print("\nIterations:")
    
    for i in range(max_iterations):
        fx = math.tan(x) - x           # f(x) = tan(x) - x
        fpx = 1.0/math.cos(x)**2 - 1   # f'(x) = sec²(x) - 1
        
        x_next = x - fx / fpx
        
        print(f"Iteration {i+1}: x = {x:.6f}, f(x) = {fx:.6f}")
        
        if abs(x_next - x) < tolerance:
            print(f"\nRoot found: {x_next:.6f}")
            break
            
        x = x_next

if __name__ == "__main__":
    newton_raphson(4.5)  # Initial guess 